### ISSUE TITLE
Can't install extensions

### ISSUE THREAD

**Issue by davidsmorais** [Ulauncher/Ulauncher#1071]
### Environment

PopO S22.04 GNOME

### Bug description

Can't install any extensions from the URL modal, here's the stack trace

```
Message: HTTP Error 403: rate limit exceeded
Error Name: UnhandledError
Type: HTTPError

Stacktrace:
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ulauncher/ui/windows/PreferencesUlauncherDialog.py", line 220, in on_scheme_callback
    resp = rt.dispatch(self, scheme_request.get_uri())
  File "/usr/lib/python3/dist-packages/ulauncher/utils/Router.py", line 49, in dispatch
    return callback(context, url_params)
  File "/usr/lib/python3/dist-packages/ulauncher/ui/windows/PreferencesUlauncherDialog.py", line 463, in prefs_extension_add
    ext_id = downloader.download(url)
  File "/usr/lib/python3/dist-packages/ulauncher/api/server/ExtensionDownloader.py", line 75, in download
    commit = gh_ext.find_compatible_version()
  File "/usr/lib/python3/dist-packages/ulauncher/api/server/GithubExtension.py", line 80, in find_compatible_version
    return self.get_commit(sha_or_branch)
  File "/usr/lib/python3/dist-packages/ulauncher/api/server/GithubExtension.py", line 89, in get_commit
    resp = json.loads(urlopen(url).read().decode('utf-8'))
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: rate limit exceeded

```


### Log output

_No response_

### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines and checked the troubleshooting page for self-help.

### Not an extension issue

- [X] This issue applies to the Ulauncher application, not a specific extension.

### Ulauncher v6

- [X] I have made sure the bug isn't already fixed in v6.

**Comment by abiriadev** [comment]
I got the same error on Arch Linux `(5.17.9-arch1-1)`

**Comment by friday** [comment]
It means that you made too many requests to GitHub within the last time period. The limit is 5000 API calls per hour.

**Comment by friday** [comment]
I just told you what the message means. This is not a matter of opinions. The error message come straight from GitHub. We don't put that in. Why you were rate limited is not really relevant, but that's why _they_ blocked your request.

**Comment by friday** [comment]
This issue was fixed in #1196 (unreleased v6 feature). This completely removes the use of the GitHub API in favor of using the same method that Git itself uses. It also avoids one request (previously two requests were needed per extension). Rate limits might still apply, but should be less aggressive.

Additionally #1042 removed the only place where these APIs would be called without the user explicitly asking for it.
